cmd/machine-controller-manager/app/controllermanager.go

```diff
@@ -141,24 +143,23 @@ func Run(ctx context.Context, s *options.MCMServer) error {
 			ClientConfig: targetkubeconfig,
 		}
 
-		err := StartControllers(
+		if err := StartControllers(
+			ctx,
 			s,
 			controlkubeconfig,
 			targetkubeconfig,
 			controlMachineClientBuilder,
 			controlCoreClientBuilder,
 			targetCoreClientBuilder,
 			recorder,
-			stop,
-		)
-
-		klog.Fatalf("error running controllers: %v", err)
-		panic("unreachable")
-
+		); err != nil {
+			klog.Fatalf("failed to start controllers: %v", err)
+		}
+		waitGroup.Done()
 	}
 
 	if !s.LeaderElection.LeaderElect {
-		run(ctx)
+		startControllers(ctx)
```

Comment:
on sending `Ctrl-C` in case when mcm runs with `leader-elect` disabled, the leader elect is attempted again
```
^CI1109 12:23:33.015676   54619 controller.go:525] Shutting down Machine Controller Manager 
I1109 12:23:33.015729   54619 reflector.go:225] Stopping reflector *v1.Node (12h0m0s) from k8s.io/client-go/informers/factory.go:134
.
.
.
I1109 12:23:33.016592   54619 reflector.go:225] Stopping reflector *v1alpha1.OpenStackMachineClass (12h0m0s) from github.com/gardener/machine-controller-manager/pkg/client/informers/externalversions/factory.go:117
I1109 12:23:35.016078   54619 leaderelection.go:248] attempting to acquire leader lease shoot--i544024--ca-test/machine-controller-manager...
E1109 12:23:35.016100   54619 controllermanager.go:193] leaderelection lost
```

a `return` here would help